akka-http-core/src/main/scala/akka/http/scaladsl/ClientTransportWithCustomResolver.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import java.net.InetSocketAddress
+
+import akka.actor.ActorSystem
+import akka.annotation.ApiMayChange
+import akka.http.scaladsl.Http.OutgoingConnection
+import akka.http.scaladsl.settings.ClientConnectionSettings
+import akka.stream.scaladsl.{ Flow, Tcp }
+import akka.util.ByteString
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+@ApiMayChange
+trait ClientTransportWithCustomResolver extends ClientTransport {
+  override def connectTo(host: String, port: Int, settings: ClientConnectionSettings)(implicit unused: ActorSystem): Flow[ByteString, ByteString, Future[Http.OutgoingConnection]] = {
+    // delay hostname resolution until stream materialization
+    Flow.setup { (mat, _) =>
```

Comment:
Do we really need the `mat` here? We've got a system which we could use for the materializer or the ec.

Reply:
We need the `Flow.setup` so resolving kicks in during materialisation. We can use the actor system parameter instead.